lib/noted/notes.ex

```diff
@@ -344,6 +345,53 @@ defmodule Noted.Notes do
     Tag.changeset(tag, attrs)
   end
 
+  def add_tag(user_id, note_id, tag_name) do
+
+    tag = Repo.get_by(Tag, name: tag_name, user_id: user_id)
+    if tag == nil do
```

Comment:
If the tag exists, it may still need us to create a NotesTags entry to form the association between tag and note.

Reply:
We should probably check if the NotesTags row exists first. If it does we can assume the tag exists.
If the NotesTags doesn't exist we check if the tag exists.
When we are done in add_tag there should either be a Tag + NotesTags entry, or an error because something is wrong.